src/plugins/discover/public/application/embeddable/saved_search_embeddable.tsx

```diff
@@ -344,64 +191,226 @@ export class SearchEmbeddable
+    searchProps.columns = handleSourceColumnState(
       { columns: this.input.columns || this.savedSearch.columns },
       this.services.core.uiSettings
     ).columns;
+
     const savedSearchSort =
       this.savedSearch.sort && this.savedSearch.sort.length
         ? this.savedSearch.sort
         : getDefaultSort(
-            this.searchScope?.indexPattern,
+            this.searchProps?.indexPattern,
             getServices().uiSettings.get(SORT_DEFAULT_ORDER_SETTING, 'desc')
           );
-    searchScope.sort = this.input.sort || savedSearchSort;
-    searchScope.sharedItemTitle = this.panelTitle;
-
+    searchProps.sort = this.input.sort || savedSearchSort;
+    searchProps.sharedItemTitle = this.panelTitle;
     if (forceFetch || isFetchRequired) {
-      this.filtersSearchSource!.setField('filter', this.input.filters);
-      this.filtersSearchSource!.setField('query', this.input.query);
+      this.filtersSearchSource.setField('filter', this.input.filters);
+      this.filtersSearchSource.setField('query', this.input.query);
       if (this.input.query?.query || this.input.filters?.length) {
-        this.filtersSearchSource!.setField('highlightAll', true);
+        this.filtersSearchSource.setField('highlightAll', true);
       } else {
-        this.filtersSearchSource!.removeField('highlightAll');
+        this.filtersSearchSource.removeField('highlightAll');
       }
 
       this.prevFilters = this.input.filters;
       this.prevQuery = this.input.query;
       this.prevTimeRange = this.input.timeRange;
       this.prevSearchSessionId = this.input.searchSessionId;
-      this.fetch();
-    } else if (this.searchScope) {
-      // trigger a digest cycle to make sure non-fetch relevant changes are propagated
-      this.searchScope.$applyAsync();
+      this.searchProps = searchProps;
+      await this.fetch();
+      if (this.node) {
+        await this.rerenderComponent(this.node);
+      }
+    } else if (this.searchProps && this.node) {
+      this.searchProps = searchProps;
+      await this.renderReactComponent(this.node, this.searchProps);
+    }
+  }
+
+  /**
+   *
+   * @param {Element} domNode
+   */
+  public async render(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      throw new Error('Search props not defined');
+    }
+    if (this.node) {
+      ReactDOM.unmountComponentAtNode(this.node);
     }
+    this.node = domNode;
+  }
+
+  private async rerenderComponent(domNode: HTMLElement) {
+    if (!this.searchProps) {
+      return;
+    }
+    this.searchProps.refs = domNode;
+    await this.pushContainerStateParamsToProps(this.searchProps);
+    await this.renderReactComponent(domNode, this.searchProps);
+  }
+
+  private async renderReactComponent(domNode: HTMLElement, searchProps: SearchProps) {
+    return new Promise(() => {
+      ReactDOM.render(<SavedSearchEmbeddableComponentMemoized {...searchProps} />, domNode);
+    });
+  }
+
+  public reload() {
+    if (this.searchProps)
```

Comment:
:art: I think we try not to skip the curly braces whenever possible, so I believe it might be a bit more aligned with the rest of the code if we'd also put that block into curly braces.